submissions/lalalilulu/friends-app/script.js

```diff
@@ -0,0 +1,109 @@
+document.addEventListener("DOMContentLoaded", () => {
+    fetchContent().then((response) => startApp(response.results)).catch(error => console.log(error.message));
+})
+
+const startApp = (response) => {
+    renderContent(response);
+    const friendsArray = response;
+
+    document.getElementById("search").addEventListener("keyup", ({target}) => {
+        renderContent(searchByValue(friendsArray, target.value));
+    });
+
+    document.querySelector(".filter-items").addEventListener("change", ({target}) => {
```

Comment:
`document.querySelector(".YOUR_FORM").addEventListener("input", ({target} => {})` instead will be enough

Reply:
**filtering and sorting still don't work correctly together. Please test your application carefully**

Reply:
Thank you for great recommendations! I have improved my code. 
Unfortunately, after the first improvement I cannot reproduce the bug with filtering and sorting – everything works fine, I suppose.

Test 1:
My test steps/ results: 
1.	Choose the gender (male in this test) – Only male friend are shown, Male Icon is chosen
2.	Enter a name for searching (Frank in this test) – Only Frank is shown, Male Icon is chosen
3.	Clear the search field - Only male friend are shown, Male Icon is chosen

Test 2:
My test steps/ results: 
1.	Choose the gender (female in this test) – Only female friend are shown, Female icon is chosen
2.	Enter a name for searching (Lena in this test) – Two friends cards are shown, Female icon is chosen
3.	Clear the search field - Only female friend are shown, Female icon is chosen

The video with tests is located in my Google Drive - https://drive.google.com/file/d/1hyW64GL3TjZD5FGWcBwsEHaNn0jgW66N/view?usp=sharing

Could you please help me with this? What I should do for bug reproducing? What result is not correct?

Reply:
1. Choose gender male.
2. Choose sorting by age. 
After these steps i expect to see men sorted by age.
But what do I really see?

Reply:
Thanks for the clarification - now I see what wrong is. 
And I hope, I have fixed filtering/sorting bug - please check the last application version